embulk-core/src/main/java/org/embulk/spi/json/JsonParser.java

```diff
@@ -29,11 +31,19 @@ public JsonParser() {
     }
 
     public Stream open(InputStream in) throws IOException {
-        return new StreamParseContext(factory, in);
+        return open(in, null);
+    }
+
+    public Stream open(InputStream in, String pointer) throws IOException {
```

Comment:
Possibly nice to have a more descriptive variable name what the `pointer` is for.

Reply:
Got it. How about `jsonPointer`, `jsonPointerExpression` or `pointerExpression`?

Reply:
I see, thank you for the detailed explanation. So I will change it to `offsetInJsonPointer`.

Reply:
How about changing the method name? (As commented, I'd suggest that way more.)

Reply:
It makes sense. So I will rename the method.